app/models/message.rb

```diff
@@ -31,6 +31,25 @@ def create_from_queue(body)
     self
   end
 
+  def set_status!
```

Comment:
We should add some validations to the status call you've just added. See `models/transmission.rb` for where we validate `#state`.

I hate to do this, can we change `#status` to `#state` here and keep it consistent with how we are doing state in the `Transmission` model. `#status` there refers to the http status code and i don't want it to be confusing.